jbi/runner.py

```diff
@@ -26,28 +26,29 @@ def execute_action(
 
     The value returned by the action call is returned.
     """
-    log_context = {
-        "bug": {
+    log_context = RunnerLogContext(
+        bug={
             "id": request.bug.id if request.bug else None,
         },
```

Comment:
Is there _actually_ a case where `request.bug` would be `None`? 

Like in `BugzillaWebhookRequest`, I think we could change bug: `Optional[BugzillaBug]` to `bug: BugzillaBug`. Then we wouldn't need to check for `request.bug`, then we wouldn't need the `BugId` type.

Or maybe if we want to keep the guard(s), we do:

```diff
diff --git a/jbi/runner.py b/jbi/runner.py
index c5a83e8..2d36ef7 100644
--- a/jbi/runner.py
+++ b/jbi/runner.py
@@ -27,9 +27,6 @@ def execute_action(
     The value returned by the action call is returned.
     """
     log_context = RunnerLogContext(
-        bug={
-            "id": request.bug.id if request.bug else None,
-        },
         request=request,
         operation=Operation.HANDLE,
     )
@@ -40,7 +37,7 @@ def execute_action(
         )
         if not request.bug:
             raise IgnoreInvalidRequestError("no bug data received")
-
+        log_context.set(bug=request.bug.dict())
         try:
             bug_obj: BugzillaBug = request.bugzilla_object
         except Exception as ex:

```

Reply:
Maybe we could return a 400 if no bug is specified instead of going that far and executing the runner... but I feel that this could be another issue/PR

Reply:
https://github.com/mozilla/jira-bugzilla-integration/issues/218